src/ctrl/nlas/mcast.rs

```diff
@@ -4,13 +4,42 @@ use crate::constants::*;
 use anyhow::Context;
 use byteorder::{ByteOrder, NativeEndian};
 use netlink_packet_utils::{
-    nla::{Nla, NlaBuffer},
+    nla::{Nla, NlaBuffer, NlasIterator, NLA_F_NESTED},
     parsers::*,
     traits::*,
     DecodeError,
 };
 use std::mem::size_of_val;
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct McastGroup {
+    pub nlas: Vec<McastGrpAttrs>,
+}
+
+impl Nla for McastGroup {
+    fn value_len(&self) -> usize {
+        self.nlas.iter().map(|nla| nla.buffer_len()).sum()
+    }
+
+    fn kind(&self) -> u16 {
+        NLA_F_NESTED
```

Comment:
The kind() is not `NLA_F_NESTED`. It is `index + 1`. The kernel code is:

```c
nla_grps = nla_nest_start_noflag(skb, CTRL_ATTR_MCAST_GROUPS);
if (nla_grps == NULL)
	goto nla_put_failure;

for (i = 0; i < family->n_mcgrps; i++) {
	struct nlattr *nest;
	const struct genl_multicast_group *grp;

	grp = &family->mcgrps[i];

	nest = nla_nest_start_noflag(skb, i + 1); // <<<<<< HERE
	if (nest == NULL)
		goto nla_put_failure;
```

Please check `BondIpAddrNlaList` in `netlink-packet-route` as an example.

Reply:
Ah, thanks for the reference. Updated based on your example.